sky/backends/backend_utils.py

```diff
@@ -1786,21 +1786,40 @@ def _update_cluster_status_no_lock(
             if returncode:
                 raise exceptions.FetchIPError(
                     reason=exceptions.FetchIPError.Reason.HEAD)
+        ray_cluster_up = True
+        # For non-spot clusters:
         # If we get node ips correctly, the cluster is UP. It is safe to
         # set the status to UP, as the `handle.external_ips` function uses ray
         # to fetch IPs and starting ray is the final step of sky launch.
+        # For spot clusters, the above can be unsafe.
```

Comment:
Can we write the reason why it is unsafe for future reference?

Reply:
I added some explanation. does it look good?